### PR TITLE
Add Datasets index page to Levelbuilder

### PR DESCRIPTION
**Pull request by ajpal**
[Part 1](https://github.com/code-dot-org/code-dot-org/pull/33139)
Fetches the list of tables according to `cdo-v3-shared/v3/channels/shared/counters/tables` and displays them as a list.
image

## Links

- [jira](https://codedotorg.atlassian.net/browse/STAR-948)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/storage/levelbuilder/DatasetList.jsx [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+  };
   render() {
-    return <h1>Datasets</h1>;
+    return (
+      <div>
+        <h1>Datasets</h1>
+        <ul>
+          {this.props.datasets.map(dataset => (
+            <li key={dataset}>
+              <a href={`/datasets/${dataset}`}>{dataset}</a>
```
Comment (jmkulwik) [inline apps/src/storage/levelbuilder/DatasetList.jsx:16]: For accessibility, should this include a title attribute? i.e. title="{dataset} dataset" --- display as ---> "Dogs dataset"

File: apps/src/storage/levelbuilder/DatasetList.jsx [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+              {this.props.liveDatasets.includes(dataset) && (
+                <span> (Live) </span>
```
Comment (jmkulwik) [inline apps/src/storage/levelbuilder/DatasetList.jsx:18]: Can you i18n this?

Comment (jmkulwik) [reply apps/src/storage/levelbuilder/DatasetList.jsx:18]: (same with other items here)

File: dashboard/app/controllers/datasets_controller.rb [hunk 3]
```diff
@@ -5,8 +5,13 @@ class DatasetsController < ApplicationController
   before_action :initialize_firebase
   authorize_resource class: false
 
+  LIVE_DATASETS = ['Daily Weather', 'Top 200 USA', 'Top 200 Worldwide', 'Viral 50 USA', 'Viral 50 Worldwide']
```
Comment (jmkulwik) [inline dashboard/app/controllers/datasets_controller.rb:8]: Is there a way to make this localized?

File: apps/src/storage/levelbuilder/DatasetList.jsx (hunk 2 above)
Comment (jmkulwik) [reply apps/src/storage/levelbuilder/DatasetList.jsx:18]: Oh, sorry, just realized this is a levelbuilder item. Nevermind about i18n

File: apps/src/storage/levelbuilder/DatasetList.jsx (hunk 1 above)
Comment (jmkulwik) [reply apps/src/storage/levelbuilder/DatasetList.jsx:16]: (also, is levelbuilder within our accessibility goals?)

File: dashboard/app/controllers/datasets_controller.rb (hunk 3 above)
Comment (ajpal) [reply dashboard/app/controllers/datasets_controller.rb:8]: What do you mean localized? These are the names of the live tables, which is not translated/translatable.

File: apps/src/storage/levelbuilder/DatasetList.jsx (hunk 1 above)
Comment (ajpal) [reply apps/src/storage/levelbuilder/DatasetList.jsx:16]: I don't think that would make this more accessible (header on the page is Datasets and the links are `/datasets/{name}` so I think it is already contextually clear that the list contains the names of datasets).

Comment (jmkulwik) [reply apps/src/storage/levelbuilder/DatasetList.jsx:16]: 👍

File: dashboard/app/controllers/datasets_controller.rb (hunk 3 above)
Comment (jmkulwik) [reply dashboard/app/controllers/datasets_controller.rb:8]: Not blocking: Is translating these titles something we'd ever need to worry about? Or if we wanted to internationalize this feature would we instead import data tables with different languages?

**Review by jmkulwik — APPROVED** [review]

**Pull request merged**